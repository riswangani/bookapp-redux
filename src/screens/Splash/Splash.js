import {
  StyleSheet,
  Text,
  View,
  ImageBackground,
  Image,
  Dimensions,
} from 'react-native';

import React, {useEffect} from 'react';

import {background, logoSplash} from '../../assets/images';

const Splash = ({navigation}) => {
  useEffect(() => {
    setTimeout(() => {
      navigation.replace('Login');
    }, 3000);
  });

  return (
    <View testID="splashScreen" style={styles.image}>
      <Image source={logoSplash} style={styles.imageLogo} />
      <Text style={styles.nama}>Riswan Gani Padilah</Text>
    </View>
  );
};

const windowHeight = Dimensions.get('window').height;

const styles = StyleSheet.create({
  image: {
    backgroundColor: '#ffffff',
    flex: 1,
    flexGrow: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
  nama: {
    top: windowHeight * 0.25,
    color: '#DB4437',
    fontWeight: '800',
    fontSize: 20,
  },
});

export default Splash;
