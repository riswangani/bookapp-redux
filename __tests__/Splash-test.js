import 'react-native';
import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import {shallow, configure} from 'enzyme';
import renderer from 'react-test-renderer';

import Splash from '../src/screens/Splash/Splash';

configure({adapter: new Adapter(), disableLifecycleMethods: true});

const tree = shallow(<Splash />);

describe('Splashscren', () => {
  it('should render correctly', () => {
    renderer.create(<Splash />);
  });

  it('should render Splash Screen, module correctly', () => {
    expect(tree).toMatchSnapshot();
  });
});
