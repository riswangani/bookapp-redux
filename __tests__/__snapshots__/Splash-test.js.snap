// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Splashscren should render Splash Screen, module correctly 1`] = `ShallowWrapper {}`;
